client/src/components/MainPage/TranslationsPage/DonateModal/AudioTranslation/index.jsx

```diff
@@ -1,11 +1,34 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faMicrophone, faStop } from '@fortawesome/free-solid-svg-icons';
 import './style.css';
 import Button from '../../../../common/Button';
 import Input from '../../../../common/Inputs';
 
 class AudioTranslation extends Component {
-   state = { selectedFile: null }
+  state = { selectedFile: null };
+
+  componentDidMount() {
+
+  }
+
+  startRecord = () => {
+    const { player } = this.refs;
+    this.setState({ recording: true });
+    const { recording } = this.state;
+    if (!recording) {
```

Comment:
@HemaSAli I can't quite understand these lines 😆
It looks like you set this.state.recording to true on line 18, then read recording back out of this.state, and if it's false, you start the actual MediaRecorder object.
Wouldn't it always be true?  I wonder if this is currently working because this.setState() doesn't actually change the state immediately, in all cases (I think sometimes it waits a moment and batches up state changes, as a performance optimization).  Either way, I think you don't need the if-statement (and it might be a source of bugs in the future, if the thing I said about setState and batching changes in the future).

Reply:
@tomduggan85 :laughing:  
In fact, I think I made a mistake here,
it must be like this
```
  const { player } = this.refs;
    const { recording } = this.state;
    if (!recording) {
    this.setState({ recording: true });
```
what I mean here is to check if it's not recording, then start recording by ``` mediaRecorder.start()```, this way is to prevent many click on start button, we can't run ``` mediaRecorder ``` many times, of course you know what I mean.
So it's just a mistake in the arrange of lines

Reply:
@HemaSAli Ah yeah, that makes sense!  Thanks for the explanation.